### DISCUSSION TITLE
umijs 支持按需 external 吗？一个集合里面用到什么再 external 什么，以及自动给 html 加入 script 引入

### DISCUSSION THREAD

**Discussion by boringplay** [umijs/umi#11687] [Q&A]
umijs 支持按需 external 吗？一个集合里面用到什么再 external 什么，以及自动给 html 加入 script 引入。

**Comment by fz6m** [comment]
所有需要 external 的依赖都需要在 [externals](https://umijs.org/docs/api/config#externals) 配置里明确指出，被 external 的依赖需要通过某种方式来提供，一般情况是 umd 。

当需要插入 script 标签内容时，需要使用 [scripts](https://umijs.org/docs/api/config#scripts) 配置。
